### PR TITLE
OCP4 STIG: Cover SRG-APP-000297-CTR-000705 with a new rule oauth_logout_url_set

### PR DESCRIPTION
**Pull request by jhrozek**
#### Description:

- Adds a new rule `oauth_logout_url_set` that checks that the console is configured
  to redirect the user to a logout page

#### Rationale:

- According to the STIG, users must have a clear way of clearing their credential

#### Review Hints:

- Make sure the tests pass
- compare the row in the autogenerated spreadsheet with [this spreadsheet](https://docs.google.com/spreadsheets/d/1oTUEUSkpumEpGqeBvhEHkJxcGUxMgoG1/edit#gid=959715005)

### CODE REVIEW

**Comment by jhrozek** [comment]
/retest
cluster installs are having a bad day

**Comment by jhrozek** [comment]
Related artifact: https://github.com/ComplianceAsCode/content/suites/10882614859/artifacts/549199083

**Comment by jhrozek** [comment]
@Vincent056 I think the remaining test failures are unrelated. PTAL

**Review by rhmdnd — APPROVED** [review]

File: applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e-remediation.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+oc patch console.config.openshift.io cluster --type merge -p '{"spec":{"authentication":{"logoutRedirect":"https://my.idp.com/logout"}}}'
```
Comment (marcusburghardt) [inline applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e-remediation.sh:3]: Could you include an empty line at the end, please?

File: applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+default_result: FAIL
+result_after_remediation: PASS
```
Comment (marcusburghardt) [inline applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e.yml:3]: Could you include an empty line at the end, please?

File: controls/srg_ctr/SRG-APP-000297-CTR-000705.yml [hunk 3]
```diff
@@ -4,4 +4,6 @@ controls:
   - low
   title: Access to the container platform must display an explicit logout message
     to user indicating the reliable termination of authenticated communication sessions.
-  status: pending
+  status: automated
+  rules:
+    - oauth_logout_url_set
```
Comment (marcusburghardt) [inline controls/srg_ctr/SRG-APP-000297-CTR-000705.yml:9]: Could you include an empty line at the end, please?

File: applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e-remediation.sh (hunk 1 above)
Comment (jhrozek) [reply applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e-remediation.sh]: done, thank you

File: applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e.yml (hunk 2 above)
Comment (jhrozek) [reply applications/openshift/general/oauth_logout_url_set/tests/ocp4/e2e.yml]: done, thank you

File: controls/srg_ctr/SRG-APP-000297-CTR-000705.yml (hunk 3 above)
Comment (jhrozek) [reply controls/srg_ctr/SRG-APP-000297-CTR-000705.yml]: done, thank you

**Comment by jhrozek** [comment]
scans just appeared to be slow
/retest

**Comment by jhrozek** [comment]
To review, you can compare the [SRG export](https://github.com/ComplianceAsCode/content/suites/11088986341/artifacts/564471345) with the current [STIG draft](https://docs.google.com/spreadsheets/d/1ouRalvXzRHuhPPN_AKMvqqscUqGvVOkkEdMP3YYBO44/edit#gid=1859516253)

**Comment by jhrozek** [comment]
Since @rhmdnd acked the patches earlier and since then I only fixed blank lines at the end of files and all CI tests are passing I'm taking the liberty to merge the PR.

**Pull request merged**